Deploy

tsohleDev opened the pull request:
Creating a Add button and remove that adds and removes the books data dynamically and preserve the data in the browser

shakespear95 committed:
added the first files

tsohleDev committed:
Add books dynamically

tsohleDev committed:
Add to local Storage

shakespear95 committed:
fixed the local storage issue

tsohleDev committed:
Add Readme file and MIT licence

tsohleDev committed:
fix linters

tsohleDev committed:
Add style sheet

tsohleDev committed:
fix lint error

tsohleDev committed:
fix lint error

ElsonOtake commented on README.md:
```diff
@@ -1 +1,283 @@
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- FUTURE FEATURES -->
+
+<!-- ## 🔭 Future Features <a name="future-features"></a>
+
+> Describe 1 - 3 features you will add to the project.
+
+- [ ] **[new_feature_1]**
+- [ ] **[new_feature_2]**
+- [ ] **[new_feature_3]** -->
+<!-- 
+<p align="right">(<a href="#readme-top">back to top</a>)</p> -->
```
- [x] Kindly add some future features for your project. It's important to show recruiters that your project is active.

tsohleDev replied:
Thank you for the very useful information 🙇

ElsonOtake commented on README.md:
```diff
@@ -1 +1,283 @@
+    - [Key Features](#key-features)
+  - [🚀 Live Demo](#live-demo)
+- [💻 Getting Started](#getting-started)
+  - [Setup](#setup)
+  - [Prerequisites](#prerequisites)
+  - [Install](#install)
+  - [Usage](#usage)
+  - [Run tests](#run-tests)
```
- [x] Please provide information about the tests of your app on local machines. It's important for recruiters and hiring managers to check your work.

tsohleDev replied:
Thank you for the information 😄

ElsonOtake commented on README.md:
```diff
@@ -1 +1,283 @@
+
+<div align="center">
+
+  <img src="https://raw.githubusercontent.com/microverseinc/readme-template/master/murple_logo.png" alt="logo" width="140"  height="auto" />
+  <br/>
```
- [x] I think it's better not to use the template logo. You can replace it with your own logo if you have one.

ElsonOtake commented on form.js:
```diff
@@ -0,0 +1,85 @@
+const books = [];
+
+function Book(title, name, node = null) {
+  this.id = books.length;
```
- [ ] I think it's better to avoid the use of `length` to create `ids`. If you have two items with id 0 and 1. And, delete the item with id 0. After that, create a new item. This new item will receive id 1, the same as the previous one.

ElsonOtake commented on form.js:
```diff
@@ -0,0 +1,85 @@
+  const BookWritter = document.createElement('p');
+  const remove = document.createElement('button');
+  remove.innerHTML = 'Remove';
+  remove.addEventListener('click', () => {
```
- [x] I think it's better to fix this remove button functionality. It's not working after deleting and recreating books. Consider using the [filter](https://www.w3schools.com/jsref/jsref_filter.asp).

ElsonOtake reviewed (changes requested):
STATUS: CHANGES REQUIRED 🟡 

Hi @tsohleDev,

Good job so far! 🎉 🎉 
There are some issues that you still need to work on to go to the next project but you are almost there! 🥲 🥲 

![woohoo-youre-almost](https://user-images.githubusercontent.com/77038610/154921145-1f7bb8f6-c843-4ba5-a534-6b840c600558.jpeg)

## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️
- PR has a good title and summary. ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

tsohleDev committed:
Fix remove button functionality

The problem was with the id of each object, since the ids could duplicate it didnt work well
I added a line that updates each objects id after removing it from the DOM

andgarzonmal reviewed (approved):
Hi @tsohleDev , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

<img src="https://media2.giphy.com/media/YKdm91XONLTaw/giphy-downsized-medium.gif"/>

## Highlights
- No linter errors ✔️ 
- Good implementations  ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
